src/tiledb/cloud/vcf/ingestion.py

```diff
@@ -0,0 +1,977 @@
+import enum
+import inspect
+import logging
+import subprocess
+from collections import defaultdict
+from math import ceil
+from typing import Any, Mapping, Optional, Sequence, Tuple, Union
+
+import numpy as np
+import tiledbvcf
+
+import tiledb.cloud
+from tiledb.cloud.utilities import Profiler
+from tiledb.cloud.utilities import create_log_array
+from tiledb.cloud.utilities import get_logger
+from tiledb.cloud.utilities import run_dag
+from tiledb.cloud.utilities import set_aws_context
+from tiledb.cloud.utilities import write_log_event
+
+LOCAL_INGEST = False
+VERBOSE = False
+TRACE = False
+
+
+def local_ingest():
+    global LOCAL_INGEST
+    LOCAL_INGEST = True
+
+
+def verbose():
+    global VERBOSE
+    VERBOSE = True
+
+
+def trace():
+    global TRACE
+    TRACE = True
+
+
+# Array names
+LOG_ARRAY = "log"
+MANIFEST_ARRAY = "manifest"
+
+# Default attributes to materialize
+DEFAULT_ATTRIBUTES = ["fmt_GT", "fmt_DP", "fmt_GQ"]
+
+# Default values for ingestion parameters
+MANIFEST_BATCH_SIZE = 100
+MANIFEST_WORKERS = 20
+VCF_BATCH_SIZE = 10
+VCF_WORKERS = 40
+VCF_THREADS = 8
+VCF_HEADER_MB = 25  # memory per sample per thread
+
+# Consolidation task resources
+CONSOLIDATE_CPUS = 4
+CONSOLIDATE_MEMORY_MB = 4096
+
+
+class Contigs(enum.Enum):
+    ALL = enum.auto()
+    CHROMOSOMES = enum.auto()
+    OTHER = enum.auto()
+
+
+def setup(
+    config: Union[Mapping[str, Any], None],
+    dataset_uri: Optional[str] = None,
+    id: Optional[str] = None,
+) -> Union[logging.Logger, Tuple[logging.Logger, Profiler]]:
```

Comment:
It's not obvious from the call under what circumstances the function will return a logger versus a logger AND a profiler.

To avoid a potential pitfall for users when using this function, I would instead make the return type one shape:

`-> Tuple[logging.Logger, Optional[Profiler]:`.